source/material_model/visco_plastic.cc

```diff
@@ -293,15 +332,31 @@ namespace aspect
       return composition_viscosities;
     }
 
+
     template <int dim>
-    void
+    std::pair<double, double>
     ViscoPlastic<dim>::
-    evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
-             MaterialModel::MaterialModelOutputs<dim> &out) const
+    calculate_weakening(const double &strain_ii,
+                        const unsigned int &j) const
```

Comment:
hand over the doubles by-value (i.e. remove the &), for simple data types like double this is equally fast (or even faster), and the chance is lower that you accidentally change them inside of the function.